src/Support/HasExtensionAttributes.php

```diff
@@ -43,10 +43,14 @@ protected function syncExtensionAttributes($attributes, $model)
     protected function resolveCategoryLinks($links, $product)
     {
         foreach ($links as $link) {
-            MagentoProductCategory::updateOrCreate([
-                'magento_product_id'  => $product->id,
-                'magento_category_id' => MagentoCategory::find($link['category_id'])->id,
```

Comment:
I assumed this was the problem part, getting an `id` of undefined or something?

Reply:
Bingo -- yeah, when the category belongs to another website for shared products but doesn't exist on the current website. Edge-case but it does happen